lxd/container_lxc.go

```diff
@@ -6883,13 +6887,155 @@ func (c *containerLXC) insertProxyDevice(devName string, m types.Device) error {
 	return nil
 }
 
+func (c *containerLXC) doNat(proxy string, device types.Device) error {
+	// Check NAT requirements
+	if device["bind"] != "host" {
+		return errors.New("NAT requirements not met")
```

Comment:
return fmt.Errorf("Only host-bound proxies can use NAT")?

Reply:
Also, this should be checked when validating the configuration rather than when applying it.